tutorial/nixify/1_nix_flake.md

```diff
@@ -0,0 +1,361 @@
+# Nixify your haskell project
+
+The purpose of this blog posts series is to simplify Haskell development workflow for you. We will begin by addressing the difficulties of working on a Haskell project without Nix, and then introduce [Nix flakes](https://nixos.wiki/wiki/Flakes) as the solution. If you're unfamiliar with Nix, we have [a quick introduction](https://haskell.flake.page/nix-rapid) available to help you get started quickly or you can take your time and explore it at [Zero to Nix](https://zero-to-nix.com). A basic understanding of the Nix expression language is assumed. Throughout the series, we will utilize a simple Haskell app called [todo-app](https://github.com/juspay/todo-app/tree/903c769d4bda0a8028fe3775415e9bdf29d80555) to illustrate how to build a Haskell project and automatically manage runtime dependencies such as databases (ie., [postgres](https://www.postgresql.org)) and and other services (here, we use [postgREST](https://postgrest.org/en/stable)), eliminating the need for any manual onboarding setup. This will allow us to highlight the advantages of using Nix.
+
+## Why Nixify?
+
+Why use Nix to develop a Haskell project rather than something like Stack or GHCup?
+
+- **Let's start with the prerequisites**: It is important to note that the Postgres server, as well as other mentioned prerequisites like postgREST, cabal-install, and GHC, do not provide the necessary information regarding their versions. Nix addresses this issue by locking these versions in `flake.lock`, ensuring that you use the same versions as the developer.
```

Comment:
I feel like this should be rewritten. Looks confusing too.

Reply:
Yes! It's about reproducibility